Microsoft.Toolkit.Uwp.UI.Controls/Carousel/Carousel.cs

```diff
@@ -0,0 +1,387 @@
+        /// </summary>
+        public double ItemRotationZ
+        {
+            get { return (double)GetValue(ItemRotationZProperty); }
+            set { SetValue(ItemRotationZProperty, value); }
+        }
+
+        // Using a DependencyProperty as the backing store for Rotation.  This enables animation, styling, binding, etc...
+        public static readonly DependencyProperty ItemRotationZProperty = DependencyProperty.Register("ItemRotationZ", typeof(double), typeof(Carousel), new PropertyMetadata(0d, OnCarouselPropertyChanged));
+
+        /// <summary>
+        /// Gets or sets the Carousel orientation. Horizontal or Vertical
+        /// </summary>
+        public Orientation Orientation
+        {
+            get { return (Orientation)GetValue(OrientationProperty); }
+            set { SetValue(OrientationProperty, value); }
+        }
+
+        // Using a DependencyProperty as the backing store for MaxViewableItems.  This enables animation, styling, binding, etc...
+        public static readonly DependencyProperty OrientationProperty = DependencyProperty.Register("Orientation", typeof(Orientation), typeof(Carousel), new PropertyMetadata(Orientation.Horizontal, OnCarouselPropertyChanged));
+
+        /// <summary>
+        /// Each time a property change, we have to reposition
+        /// </summary>
+        private static void OnCarouselPropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            if (e.NewValue == null)
+            {
+                return;
+            }
+
+            Carousel carouselControl = (Carousel)d;
+            if (carouselControl.ItemsPanel == null)
+            {
+                return;
+            }
+
+            var itemsPanel = carouselControl.GetItemsPanel();
+
+            if (itemsPanel == null)
+            {
+                return;
+            }
+
+            itemsPanel.UpdatePosition();
+        }
+
+        /// <summary>
+        /// Return ItemsPanel
+        /// </summary>
+        /// <returns>Gets the ItemsPanel used for the carousel</returns>
+        internal CarouselPanel GetItemsPanel()
+        {
+            return this.FindDescendant<CarouselPanel>();
+        }
+
+        /// <summary>
+        /// Returns the container used for each item
+        /// </summary>
+        /// <returns>Returns always a ContentControl</returns>
+        protected override DependencyObject GetContainerForItemOverride()
+        {
+            return new ContentControl();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Carousel"/> class.
+        /// </summary>
+        public Carousel()
+        {
+            // Set style
+            this.DefaultStyleKey = typeof(Carousel);
+            this.ManipulationMode = ManipulationModes.All;
+            this.IsHitTestVisible = true;
+
+            this.SubscribePropertyChanged("ItemsSource", OnCarouselPropertyChanged);
```

Comment:
Use RegisterPropertyChangedCallback method to handle the ItemsSource changing, or listen to the Items.VectorChanged event to know when items have been added/removed/changed